src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -3548,6 +3549,53 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  address generate_nmethod_entry_barrier() {
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "nmethod_entry_barrier");
+
+    address stub_address = __ pc();
```

Comment:
Please use __ function_entry(); instead of __ pc();
Otherwise it's broken on Big Endian which use ABI v1.

Reply:
Sorry, __ pc() seems to be correct. It's directly called from nmethod_entry_barrier without using a function descriptor.